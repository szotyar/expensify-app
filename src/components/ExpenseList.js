import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
  <div>
    <h1>ExpenseList</h1>
    {
      props.expenses.map((expense) => (
      <ExpenseListItem key={expense.id} {...expense} />
    )) 
    }
  </div>
); 

// const ConnectedExpenseList = connect((state) => {
//   return {
//     expenses: state.expenses
//   };
// })(ExpenseList);

// export default ConnectedExpenseList;

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
}

export default connect(mapStateToProps)(ExpenseList);