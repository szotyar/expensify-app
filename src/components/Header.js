import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" >Home</NavLink>
    <NavLink to="/create" >Create</NavLink>
    <NavLink to="/help" >Help</NavLink>
  </header>
);

export default Header;