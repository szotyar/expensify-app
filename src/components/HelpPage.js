import React from 'react';

const HelpPage = () => (
  <div>
    This is from my helppage component
  </div>
);

export default HelpPage;