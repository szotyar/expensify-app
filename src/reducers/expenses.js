//Expenses reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch(action.type){
    case 'ADD_EXPENSE':
      return [
        ...state, // spread operator
        action.expense
      ];
    case 'REMOVE_EXPENSE':
      return state.filter((item) => {
        return action.expense.id !== item.id
      });
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if(expense.id === action.id){
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });

    default:
      return state;
  }
};