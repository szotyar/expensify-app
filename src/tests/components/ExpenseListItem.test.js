import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('I would like ExpenseListItem to render', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});