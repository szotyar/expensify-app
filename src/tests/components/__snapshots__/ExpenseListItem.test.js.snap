// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I would like ExpenseListItem to render 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.23
    -
    January 1st, 1970
  </p>
</div>
`;
